src/protocol/requests.rs

```diff
@@ -138,7 +139,30 @@ pub struct CompletionParams {
 #[derive(Serialize, Deserialize, Clone)]
 pub struct CompletionContext {
     #[serde(rename = "triggerKind")]
-    trigger_kind: i32,
+    pub trigger_kind: i32,
     #[serde(rename = "triggerCharacter")]
-    trigger_character: Option<String>,
+    pub trigger_character: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct SignatureHelpParams {
+    #[serde(rename = "textDocument")]
+    pub text_document: TextDocumentIdentifier,
+    pub position: Position,
+    pub context: Option<SignatureHelpContext>,
```

Comment:
there is an additional parameters in the specs
```
        /**
	 * An optional token that a server can use to report work done progress.
	 */
	workDoneToken?: ProgressToken;
```

Reply:
LSP clients don't seem to be sending this and since its optional I opted to omit it for the time.